pymongo/operations.py

```diff
@@ -32,6 +32,15 @@ def __init__(self, filter=None, doc=None, upsert=None):
         self._doc = doc
         self._upsert = upsert
 
+    def __eq__(self, other):
+        if type(other) == type(self):
+            return (other._filter, other._doc, other._upsert) == \
+                   (self._filter, self._doc, self._upsert)
+        return False
```

Comment:
Oh, I see what you are doing here. You want to return NotImplemented if the comparison is to a type that doesn't inherit from _WriteOp and False if the type doesn't match.